driver/src/main/scala/api/collections/BulkOps.scala

```diff
@@ -65,11 +66,12 @@ private[reactivemongo] object BulkOps {
       case Some(doc) => {
         val bsz = sz(doc)
 
-        if (bsz > maxBsonSize) {
-          Left(s"size of document #${offset + docs} exceed the maxBsonSize: $bsz > $maxBsonSize")
+        // Total minimal size is key '1' size (1 byte) + type prefix (2 bytes)
+        if (bsz + 1 + BSONElementSet.typePrefixByteSize > maxBsonSize) {
+          Left(s"size of document #${offset + docs} exceed the maxBsonSize: $bsz + 3 > $maxBsonSize")
         } else {
           val nc = docs + 1
-          val nsz = bsonSize + bsz
+          val nsz = bsonSize + bsz + docs.toString.getBytes.length + BSONElementSet.typePrefixByteSize
```

Comment:
```suggestion
          val keySize = docs.toString.getBytes.size // string repr. of current index used as key
          val nsz = bsonSize + bsz + keySize + BSONElementSet.typePrefixByteSize
```

Extract as `val` to comment the tricky part about key